modules/react-native-wallet/android/src/main/java/com/wallet/WalletModule.kt

```diff
@@ -0,0 +1,304 @@
+        val hardwareId = async { getHardwareIdAsync() }
+        val walletData = WalletData(
+          platform = "android", deviceID = hardwareId.await(), walletAccountID = walletId.await()
+        )
+
+        val walletDataJson = JSONObject().apply {
+          put("platform", walletData.platform)
+          put("deviceID", walletData.deviceID)
+          put("walletAccountID", walletData.walletAccountID)
+        }
+
+        promise.resolve(walletDataJson.toString())
+      } catch (e: Exception) {
+        promise.reject("Error", "Failed to retrieve IDs: ${e.localizedMessage}")
+      }
+    }
+  }
+
+  private fun getCardStatusCode(code: Int): Int {
+    return when (code) {
+      TapAndPay.TOKEN_STATE_ACTIVE -> CardStatus.ACTIVE.code
+      TapAndPay.TOKEN_STATE_PENDING -> CardStatus.PENDING.code
+      TapAndPay.TOKEN_STATE_SUSPENDED -> CardStatus.SUSPENDED.code
+      TapAndPay.TOKEN_STATE_NEEDS_IDENTITY_VERIFICATION -> CardStatus.REQUIRE_IDENTITY_VERIFICATION.code
+      TapAndPay.TOKEN_STATE_FELICA_PENDING_PROVISIONING -> CardStatus.PENDING.code
+      else -> CardStatus.NOT_FOUND_IN_WALLET.code
+    }
+  }
+
+
+  @ReactMethod
+  override fun getCardStatus(last4Digits: String, promise: Promise) {
+    if (!ensureTapAndPayClientInitialized()) {
+      promise.reject("Initialization error", "TapAndPay client initialization failed")
+      return
+    }
+
+    tapAndPayClient!!.listTokens()
+      .addOnCompleteListener { task ->
+        if (!task.isSuccessful || task.result == null) {
+          promise.reject("error", "no tokens available")
+          return@addOnCompleteListener
+        }
+        val token = task.result.find { it.fpanLastFour == last4Digits }
+        token?.let {
+          Log.i("getCardStatus", "Card Token State: ${it.tokenState}")
+          promise.resolve(
+            getCardStatusCode(token.tokenState)
+          )
+        } ?:  promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
+      }
+      .addOnFailureListener { e -> promise.reject("getCardStatus function failed", e) }
+      .addOnCanceledListener {
+        promise.reject(
+          "Reject",
+          "Card status retrieval canceled"
+        )
+      }
+  }
+
+  @ReactMethod
+  override fun getCardTokenStatus(tsp: String, tokenRefId: String, promise: Promise) {
+    if (!ensureTapAndPayClientInitialized()) {
+      promise.reject("Initialization error", "TapAndPay client initialization failed")
+      return
+    }
+
+    tapAndPayClient!!.getTokenStatus(getTokenServiceProvider(tsp), tokenRefId)
+      .addOnCompleteListener { task ->
+        if (!task.isSuccessful || task.result == null) {
+          promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
+          return@addOnCompleteListener
+        }
+        val token = task.result
+        token?.let {
+          promise.resolve(
+            getCardStatusCode(token.tokenState)
+          )
+        } ?:  promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
```

Comment:
```suggestion
        task.result?.let { token ->
          promise.resolve(
            getCardStatusCode(token.tokenState)
          )
        } ?: promise.resolve(CardStatus.NOT_FOUND_IN_WALLET.code)
```

Not sure about this one